File: MyungJi/231127_week/programmers_42746_가장큰수.cpp
#include <bits/stdc++.h>

using namespace std;
vector<string> str;

bool cmp(string a, string b){
    return a+b > b+a;
}
string solution(vector<int> numbers) {
    string answer = "";
    //1. 문자열로 바꾸기
    for(int n:numbers) str.push_back(to_string(n));
    //2. 정렬
    sort(str.begin(), str.end(), cmp);
    //3. 정답 출력
    if(str[0]=="0") return "0";
    for(string s:str) answer+=s;
    
    return answer;
}

File: MinJu/231204_week/Programmers_42746_가장 큰 수.py
def solution(numbers):
    numbers = list(map(str, numbers)) # 문자열 리스트로 
    # 3과 31 중 3이 앞에 와야 큰 수가 됨 -> 33 31 비교 
    # 입력되는 수 1000 이하, 사전순 비교 -> 문자열 3번 반복 비교 -> 3자릿수까지 사전 비교 가능 333 323232
    numbers.sort(key=lambda x:x*3, reverse=True) # 더 큰 숫자 문자열이 앞에 나오게 내림차순 
    return str(int(''.join(numbers))) # int로 바꾸고 str 변환 (00 경우 0으로 반환하기 위해)
